header-footer-grid/Core/Builder/Abstract_Builder.php

```diff
@@ -1862,7 +1863,7 @@ public function register_component( $component_to_add ) {
 	 *
 	 * @param string|null $id The id of the component.
 	 *
-	 * @return Abstract_Component
+	 * @return Abstract_Component|null
 	 * @since   1.0.0
 	 * @access  public
 	 */
```

Comment:
We have a guard for added for this issue: https://github.com/Codeinwp/neve/issues/3557 here: https://github.com/Codeinwp/neve/blob/master/header-footer-grid/functions-template.php#L95-L97 will evaluate as `Expression in empty() is not falsy.`. Since we guard for a case where the returned component can be null I added the proper type. Another option would have been to remove the guard. I chose to change the possible type.

Reply:
Got it, thanks for the explanation. It makes sense, I think we can keep as it is, but you know better.